code/modules/admin/verbs/adminweather.dm

```diff
@@ -0,0 +1,90 @@
+ADMIN_VERB(run_weather, R_ADMIN|R_FUN, "Run Weather", "Triggers specific weather on the z-level you choose.", ADMIN_CATEGORY_WEATHER)
+
+	var/list/weather_choices = list()
+	if(!length(weather_choices))
+		for(var/datum/weather/weather_type as anything in subtypesof(/datum/weather))
+			weather_choices[initial(weather_type.type)] = weather_type
+
+	var/datum/weather/weather_choice = tgui_input_list(user, "Choose a weather to run", "Weather", weather_choices)
+	if(!weather_choice)
+		return
+	weather_choice = weather_choices[weather_choice]
+
+	var/turf/current_turf = get_turf(user.mob)
+	var/z_level = tgui_input_number(user, "Z-Level to target", "Z-Level", min_value = 1, max_value = world.maxz, default = current_turf?.z)
+	if(!isnum(z_level))
+		return
+
+	var/static/list/custom_options = list("Default", "Custom", "Cancel")
+	var/custom_choice = tgui_alert(user, "How would you like to run the weather settings?", "Custom Weather", custom_options)
+	switch(custom_choice)
+		if("Default")
+			SSweather.run_weather(weather_choice, z_level) // default settings
```

Comment:
Doesn't log that admin used this verb, needs fixing - your logging code is right at the bottom

Reply:
Applied via https://github.com/tgstation/tgstation/pull/89550/commits/8fcf3c1a6af197d680b26181dfbc31e4c20d9e41